api/v1alpha1/bundle_types.go

```diff
@@ -18,48 +18,73 @@ package v1alpha1
 
 import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	runtime "k8s.io/apimachinery/pkg/runtime"
 )
 
-// Source locates content to stage.
-type Source string
+type ProvisionerID string
+type BundleConditionType string
+
+const (
+	TypeUnpacked = "Unpacked"
+
+	ReasonUnpackPending    = "UnpackPending"
+	ReasonUnpacking        = "Unpacking"
+	ReasonUnpackSuccessful = "UnpackSuccessful"
+	ReasonUnpackFailed     = "UnpackFailed"
+
+	PhasePending   = "Pending"
+	PhaseUnpacking = "Unpacking"
+	PhaseFailing   = "Failing"
+	PhaseUnpacked  = "Unpacked"
+)
 
 // BundleSpec defines the desired state of Bundle
 type BundleSpec struct {
-	// Class specifies the name of the provisioner that should manage the Bundle.
-	// +optional
-	Class string `json:"class,omitempty"`
+	// ProvisionerClassName sets the name of the provisioner that should reconcile this BundleInstance.
+	ProvisionerClassName string `json:"provisionerClassName"`
 
-	// Source locates all remote content to be staged by the Bundle.
-	Source Source `json:"source"`
-}
+	// Image is the bundle image that backs the content of this bundle.
+	Image string `json:"image"`
```

Comment:
This field has changed from a more nebulous `Source` to a concrete `Image` field referring to container images only. Maybe we can revisit this change once other content sources becomes supported?

I kind of like the more generic term but agree with keeping things simple for now.

Reply:
Yep - I missed calling this out. Looking at the e2e strawman, it looks like we would like to support multiple `sources` over time, but we decided on keeping the API fairly scoped in the short term, which means the following:

- We only support a single Bundle source;
- We only support a remote container image Bundle source;

With that in mind, we can simply land on a spec.Image field in the short term, and work towards a model that supports multiple, different Bundle sources depending on the individual provisioner and what that provisioner supports under-the-hood.